Normalize `/sites` sidebar animation on desktop.

StevenDufresne opened the pull request:
Related to #92911

## Why are these changes being made?

When you click through the sidebar items, you'll get differing animations in the sidebar. 

## Proposed Changes

At face value, this seems like a problem with the transition and, it kinda is I guess. 

We switch out `--sidebar-width-max` from `272px` to `295px` once we add [.is-global-sidebar-visible](https://github.com/Automattic/wp-calypso/blob/b1f6370587d155dc14b40ef89d5b7a46f56a946f/client/my-sites/sidebar/style.scss#L15) in [client/layout/index.jsx](https://github.com/Automattic/wp-calypso/blob/b1f6370587d155dc14b40ef89d5b7a46f56a946f/client/layout/index.jsx#L258).

With Domains and Themes there is a delay in `isGlobalSidebarVisible` being said to true. That delay is why we see the transition. 

The solution in this PR is to add the routes to `getSiteDashboardRoutes` which is used by [shouldShowSitesDashboard]( https://github.com/Automattic/wp-calypso/blob/b1f6370587d155dc14b40ef89d5b7a46f56a946f/client/state/global-sidebar/selectors.ts#L39) to determine (earlier) that we are viewing a page with the global sidebar.


## Considerations

I don't completely understand the implication of adding the routes here. The comment suggests that apart from affecting this layout, the presence of those routes will determine where they should up. Reading through the associated p2 also leaves things unclear for me. 

Maybe you can confirm or provide better context @fushar? (Updated [here](https://github.com/Automattic/wp-calypso/pull/101154/files#diff-33316ee02ed424c6862ab81c9792fd225a851305558dca401c9029a2dc2f94fbR12)).  


### Screenshots

| Before | After |
|--------|--------|
| ![Screen Capture on 2025-03-26 at 11-20-33](https://github.com/user-attachments/assets/21bf1a0c-e439-42bb-9247-802ee371ac2c) | ![Screen Capture on 2025-03-26 at 11-22-09](https://github.com/user-attachments/assets/cc6bc355-9e70-4559-a886-e91c265863c0) | 

## Testing Instructions
- Go to `/sites`
- Click on "Domains" & "Themes"
- You should not see any width transition in the sidebar.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

StevenDufresne committed:
Add the missing routes.

fushar reviewed (approved):
Thanks for this! I believe it is an acceptable solution for now given the current way on how we do things.

There's more context and discussion here: paYKcK-5Qk-p2

StevenDufresne committed:
Add tests.

StevenDufresne committed:
Separate out the tangled base paths.

StevenDufresne commented:
@fushar Thanks for the quick review. 

Luckily an e2e test failed and exposed an error in the original logic. Because we're using `startsWith` to determine if a route matches, if a user is using the default WordPress.com interface and clicks "Appearance > Themes", the URL would be `/themes/{domain}` and they would be re-routed to the `sites-dashboard` even though they are in their "sites" context. Same problem with `/domains/manage/`. I've broken out those routes and moved them into the `shouldShowSitesDashboard`. I also moved them into a function called `tangledBasePaths` to communicate that they are still used in both contexts and added some user tests.

StevenDufresne committed:
Fix typo.

fushar reviewed (approved):
Oohhh.... good catch 🤦 thanks, I've smoke-tested and didn't find any regression 👍
